scripts/ci/payload-generator/main.go

```diff
@@ -15,6 +15,9 @@ import (
 )
 
 var (
+	defaultInputDir  = filepath.Join(environment.GetRootPathUnsafe(), "build")
+	defaultOutputDir = filepath.Join(defaultInputDir, "payload", "state-install")
```

Comment:
I'd rather not bury it this deep. It should just be `build/<dir>`. I'm guessing this is for the generate-test-update logic? Suggest using `build/test-payload` for that, and `build/payload` here.

Reply:
In general, the archive is expected to be in a directory called "state-install". So, we can shorten this to build/state-install, or the update generator can copy the files from whatever directory we want into a temp directory "{tmp}/state-install", so the final archive is correct.

Reply:
@Naatan I think this still could use your guidance.

Reply:
Where is this requirement of it having to be in a directory named "state-install" coming from? It sounds like whatever is creating this requirement needs to be updated.